### PR TITLE
[INLONG-5047][Agent] Support collect data from SQLServer

### LINKED ISSUE #5047: [Feature][Agent] Support collect data from SQLServer
### Description

you can refer to:
https://inlong.apache.org/docs/next/design_and_concept/how_to_write_plugin_agent

### Use case

_No response_

### Are you willing to submit PR?

- [ ] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by haibo-duan**
### Prepare a Pull Request
*(Change the title refer to the following example)*

- Title Example: [INLONG-XYZ][Component] Title of the pull request

*(The following *XYZ* should be replaced by the actual [GitHub Issue](https://github.com/apache/inlong/issues) number)*

- Fixes #5047 

### Motivation

*Explain here the context, and why you're making that change. What is the problem you're trying to solve?*

### Modifications

*Describe the modifications you've done.*

### Verifying this change

*(Please pick either of the following options)*

- [ ] This change is a trivial rework/code cleanup without any test coverage.

- [ ] This change is already covered by existing tests, such as:
  *(please describe tests)*

- [ ] This change added tests and can be verified as follows:

  *(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a follow-up issue for adding the documentation

### CODE REVIEW

File: inlong-agent/agent-plugins/src/test/java/org/apache/inlong/agent/plugin/sources/TestSQLServerSource.java [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+        // build mock
+        when(jobProfile.get(eq(CommonConstants.PROXY_INLONG_GROUP_ID), anyString())).thenReturn("test_group");
+        when(jobProfile.get(eq(CommonConstants.PROXY_INLONG_STREAM_ID), anyString())).thenReturn("test_stream");
+        when(jobProfile.get(eq(SQLServerSource.JOB_DATABASE_SQL), eq(StringUtils.EMPTY))).thenReturn(StringUtils.EMPTY,
+                sql1, sql2);
+
+        final SQLServerSource source = new SQLServerSource();
+
+        //assert
+        assertEquals(null, source.split(jobProfile));
```
Comment (healchow) [inline inlong-agent/agent-plugins/src/test/java/org/apache/inlong/agent/plugin/sources/TestSQLServerSource.java:99]: Add one blank after `//`.

File: inlong-agent/agent-plugins/src/test/java/org/apache/inlong/agent/plugin/sources/TestSQLServerSource.java [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+        PowerMockito.doNothing().when(
+                MetricRegister.class, "register", any(MetricItem.class));
+    }
+
+    /**
+     * Test cases for {@link SQLServerSource#split(JobProfile)}.
+     */
+    @Test
+    public void testSplit() {
+
```
Comment (healchow) [inline inlong-agent/agent-plugins/src/test/java/org/apache/inlong/agent/plugin/sources/TestSQLServerSource.java:86]: Unnecessary blank line.

File: inlong-agent/agent-plugins/src/test/java/org/apache/inlong/agent/plugin/sources/TestSQLServerReader.java [hunk 3]
```diff
@@ -0,0 +1,257 @@
...
+        reader.init(jobProfile);
+    }
+
+
+    /**
+     * Test cases for {@link SQLServerReader#read()}.
+     */
+    @Test
+    public void testRead() throws Exception {
+
```
Comment (healchow) [inline inlong-agent/agent-plugins/src/test/java/org/apache/inlong/agent/plugin/sources/TestSQLServerReader.java:162]: Unnecessary blank line.

File: inlong-agent/agent-plugins/src/test/java/org/apache/inlong/agent/plugin/sources/TestSQLServerReader.java [hunk 4]
```diff
@@ -0,0 +1,257 @@
...
+        //mock metrics
+        whenNew(AgentMetricItemSet.class).withArguments(anyString()).thenReturn(agentMetricItemSet);
+        when(agentMetricItemSet.findMetricItem(any())).thenReturn(agentMetricItem);
+        field(AgentMetricItem.class, "pluginReadCount").set(agentMetricItem, atomicLong);
+
+        //init method
+        reader = new SQLServerReader(sql);
+        reader.init(jobProfile);
+    }
+
```
Comment (healchow) [inline inlong-agent/agent-plugins/src/test/java/org/apache/inlong/agent/plugin/sources/TestSQLServerReader.java:155]: Unnecessary more than one line.

File: inlong-agent/agent-plugins/src/test/java/org/apache/inlong/agent/plugin/sources/TestSQLServerReader.java [hunk 5]
```diff
@@ -0,0 +1,257 @@
...
+        when(resultSet.getMetaData()).thenReturn(metaData);
+        when(metaData.getColumnCount()).thenReturn(2);
+        when(metaData.getColumnName(1)).thenReturn("id");
+        when(metaData.getColumnName(2)).thenReturn("cell");
+        when(metaData.getColumnType(1)).thenReturn(Types.INTEGER);
+        when(metaData.getColumnType(2)).thenReturn(Types.VARCHAR);
+        when(metaData.getColumnTypeName(1)).thenReturn(typeName1);
+        when(metaData.getColumnTypeName(2)).thenReturn(typeName2);
+
+        //mock metrics
```
Comment (healchow) [inline inlong-agent/agent-plugins/src/test/java/org/apache/inlong/agent/plugin/sources/TestSQLServerReader.java:146]: Add one blank after `//`.

File: inlong-agent/agent-plugins/src/test/java/org/apache/inlong/agent/plugin/sources/TestSQLServerReader.java [hunk 6]
```diff
@@ -0,0 +1,257 @@
...
+
+    private AtomicLong atomicLong;
+
+    private String tagName;
+
+    private String sql;
+
+    @Before
+    public void setUp() throws Exception {
+
```
Comment (healchow) [inline inlong-agent/agent-plugins/src/test/java/org/apache/inlong/agent/plugin/sources/TestSQLServerReader.java:102]: Unnecessary blank line.

File: inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sources/reader/SQLServerReader.java [hunk 7]
```diff
@@ -0,0 +1,222 @@
...
+    private String separator;
+
+    public SQLServerReader(String sql) {
+        this.sql = sql;
+    }
+
+    @Override
+    public Message read() {
+        try {
+            if (resultSet.next()) {
```
Comment (healchow) [inline inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sources/reader/SQLServerReader.java:93]: Is it possible to use guard statements to reduce the nesting of code here?

File: inlong-agent/agent-plugins/src/test/java/org/apache/inlong/agent/plugin/sources/TestSQLServerSource.java [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+    private AtomicLong sourceSuccessCount;
+
+    private AtomicLong sourceFailCount;
+
+    @Before
+    public void setup() throws Exception {
+        sourceSuccessCount = new AtomicLong(0);
+        sourceFailCount = new AtomicLong(0);
+        //mock metrics
```
Comment (healchow) [inline inlong-agent/agent-plugins/src/test/java/org/apache/inlong/agent/plugin/sources/TestSQLServerSource.java:71]: Add one blank after `//`.

File: inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sources/SQLServerSource.java [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+import org.apache.inlong.agent.utils.AgentDbUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+
+public class SQLServerSource extends AbstractSource {
```
Comment (EMsnap) [inline inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sources/SQLServerSource.java:34]: add java doc pls

Comment (haibo-duan) [reply inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sources/SQLServerSource.java:37]: OK

**Review by healchow — APPROVED** [review]

**Review by dockerzhang — APPROVED** [review]

**Review by EMsnap — APPROVED** [review]

**Pull request merged**